src/control/system/raft/database.go

```diff
@@ -960,3 +965,45 @@ func (db *Database) OnEvent(_ context.Context, evt *events.RASEvent) {
 		db.handlePoolRepsUpdate(evt)
 	}
 }
+
+// SetSystemAttrs submits an update to the system properties map.
+func (db *Database) SetSystemAttrs(props map[string]string) error {
+	if err := db.CheckLeader(); err != nil {
+		return err
+	}
+	db.Lock()
+	defer db.Unlock()
+
+	if err := db.submitSystemAttrsUpdate(props); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetSystemAttrs returns a copy of the system properties map.
+func (db *Database) GetSystemAttrs(keys []string) (map[string]string, error) {
+	if err := db.CheckReplica(); err != nil {
+		return nil, err
+	}
+	db.data.RLock()
+	defer db.data.RUnlock()
+
+	// Always return a copy of the map to avoid safety issues.
+	out := make(map[string]string)
+	if len(keys) == 0 {
+		for k, v := range db.data.System.Attributes {
+			out[k] = v
+		}
+		return out, nil
+	}
+
+	for _, k := range keys {
```

Comment:
why not just return a copy of the System.Attributes? what does iterating through the keys and values get you?

Reply:
`System.Attributes` is a map, and this is how a map is copied. The `copy()` built-in only works for slices.